jsonargparse/_parameter_resolvers.py

```diff
@@ -855,6 +892,8 @@ def get_parameters_by_assumptions(
 ) -> ParamList:
     component, parent, method_name = get_component_and_parent(function_or_class, method_name)
     params, args_idx, kwargs_idx, _, stubs = get_signature_parameters_and_indexes(component, parent, logger)
+    if unpack_typed_dict_kwargs(params):
+        kwargs_idx = -1
```

Comment:
Why do you want to support the unpack in the resolver by assumptions? Long back this was the only resolver and it was kept for backwards compatibility of some internal code we had. But apart from us, I doubt there are people that need this. Do you need it?

If we don't know of anyone who needs it, I would prefer to not have it. Whoever needs it could create a feature request in the future.

Reply:
I wasn't aware it was practically deprecated, so I added this test case to ensure test coverage of changed lines. It happens that I cannot find any other way to make it get invoked without an unrelated error, so I will do as you suggest and remove unpack in the resolver by assumptions.